include/rocksdb/file_system.h

```diff
@@ -640,6 +645,15 @@ class FileSystem : public Customizable {
                                const IOOptions& options, bool* is_dir,
                                IODebugContext* /*dgb*/) = 0;
 
+  // Poll for completion of read IO. The Poll() method should call the callback
+  // provided to ReadAsync to indicate completion. A return value of
+  // IOStatus::NotSupported() indicates that polling is not supported, and
```

Comment:
It might make sense to make supporting Poll and ReadAsync mandatory. When implementing the feature, we could provide a default implementation that's synchronous, and the FileSystem implementation could override it to make it async. That'll simplify the RocksDB code.

Reply:
Sure. I will update the `ReadAsync`. For `Poll` API, Should I return with `Status::OK` as default Read was synchronous?

Reply:
Yes, I think that's appropriate